ion-schema/src/main/kotlin/com/amazon/ionschema/model/AstSchema.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model
+
+import com.amazon.ionelement.api.IonElement
+
+/**
+ * Represents a complete Ion Schema document. Note—the model does not include a particular ISL version because the AST
+ * is intended to be a version-agnostic representation of ISL.
+ */
```

Comment:
So how can you be sure the AST will be version-agnostic? E.g. what if a future ISL version were to rename a constraint?

Reply:
I am not _sure_ that it always can be, but it is my intention. I do have a few thoughts about how to handle changing/adding/removing constraints:

* The AST does not need to have a 1:1 mapping with any ISL keywords. The goal is accurately representing the function/behavior of a type definition.
* It is possible to model a constraint that has more than one ID with this AST. If ISL 3.0 changes the name of `valid_values` to `ok_values`, we can just create a new ID:  `val OkValues = ConstraintId<ValidValuesConstraint>("ok_values")`.
* Suppose we have a `foo` constraint that is only in ISL 3.0 and a `bar` constraint that is only in ISL 4.0, it would be possible to have both a `FooConstraint` and a `BarConstraint` in the AST. If you include both of them in a schema, the serializer may reject it as being invalid, but it is _possible_ for them to co-exist in this AST.